### PR TITLE
refactor(desktop): split task approval flow orchestration

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- split `useTaskApprovalFlow` into a reducer-backed modal state model plus focused direct-merge, pull-request, and git-conflict helpers
- preserve the existing Kanban approval behavior while making stale open-cycle handling and direct-merge completion transitions explicit
- expand approval-flow coverage for stale responses, manual PR success, conflict recovery branches, and direct-merge completion recoverability

## Verification
- `bun run --filter @openducktor/desktop typecheck`
- `bun run --filter @openducktor/desktop test`

## Summary by CodeRabbit

* **New Features**
  * Enhanced task approval workflows with direct merge capability and conflict detection.
  * Added pull request creation within the approval process.
  * Improved git conflict handling and resolution in task approvals.

* **Refactor**
  * Restructured task approval state management for improved maintainability.

* **Tests**
  * Added comprehensive test coverage for task approval workflows and edge cases.

### CODE REVIEW

File: apps/desktop/src/pages/kanban/task-approval-flow-state.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    case "load_succeeded":
+      return buildTaskApprovalLoadedState(action);
+    case "close":
+      return CLOSED_TASK_APPROVAL_STATE;
+    case "start_submitting":
+      return state.kind === "open" ? { ...state, phase: "submitting" } : state;
+    case "return_to_editable":
+      return state.kind === "open"
+        ? { ...state, phase: "ready", errorMessage: action.errorMessage }
+        : state;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/task-approval-flow-state.ts:136]: Fixed in `8d446022` and covered by `20f1a1b4`. `return_to_editable` now only transitions an actively submitting modal, so a stale completion from task A can no longer unlock a newer loading modal for task B.

File: apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    };
+
+const normalizeConflict = (
+  conflict: NonNullable<
+    Awaited<ReturnType<typeof host.taskDirectMerge>> extends infer TResult
+      ? TResult extends { outcome: "conflicts"; conflict: infer TConflict }
+        ? TConflict
+        : never
+      : never
+  >,
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts:48]: Keeping the current conditional type here. It already derives the conflict payload from `host.taskDirectMerge` without casts or duplicated domain typing, and switching this spot to `Extract<>` would be a style-only rewrite rather than a behavioral or safety improvement.

File: apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  if (publishTarget) {
+    if (!publishTarget.remote) {
+      throw new Error("The configured target branch does not have a publish remote.");
+    }
+    const pushResult = await host.gitPushBranch(repoPath, checkoutTargetBranch(publishTarget), {
+      remote: publishTarget.remote,
+    });
+    if (pushResult.outcome !== "pushed") {
+      throw new Error(pushResult.output);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts]: I fixed the empty-message case in `8d446022`, but I am not adding a one-off custom error class here. This frontend layer currently reports actionable strings through `errorMessage(...)`, and there is not an established custom error-class/`instanceof` pattern in this area to extend.

File: apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    }
+  }
+
+  await host.taskDirectMergeComplete(repoPath, approval.taskId);
+  await refreshTasks();
+
+  return {
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts:126]: Fixed in `8d446022`. Direct-merge publish failures now fall back to `Git push failed with no output.` when the host returns an empty message.

File: apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      successDescription: canonicalTargetBranch(approvalContext.targetBranch),
+    };
+  }
+
+  await invalidateTaskApprovalContextQuery(queryClient, repoPath, approval.taskId);
+  const nextApprovalContext = await loadTaskApprovalContextFromQuery(
+    queryClient,
+    repoPath,
+    approval.taskId,
+  );
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/task-approval-flow-direct-merge.ts:90]: I am not taking this change as written. The suggested best-effort `refreshTasks()` fallback would mask a broken post-mutation sync path, and this repo explicitly forbids fallback logic that hides failures. Here the host mutation is authoritative, but the follow-up refresh/query work is still required to rehydrate the UI correctly; if that sync fails we want an actionable error rather than silently continuing with stale task state. If we decide to improve this later, it should be done as a more explicit success-with-warning design across the approval flow rather than a local swallow here.
